support/metrics/metricsTracker.go

```diff
@@ -24,13 +24,14 @@ const (
 // and can be used to directly send events to the
 // Amplitude HTTP API.
 type MetricsTracker struct {
-	client     *http.Client
-	apiKey     string
-	userID     string
-	deviceID   string
-	props      commonProps
-	start      time.Time
-	isDisabled bool
+	client               *http.Client
+	apiKey               string
+	userID               string
+	deviceID             string
+	props                commonProps
+	botStartTime         time.Time
+	isDisabled           bool
+	LastMetricUpdateTime time.Time
```

Comment:
This should be unexported and should have an exported Get method. See second comment to get full pattern of usage.

Also, the name is very generic now that it’s on the struct. It’s not clear if this represents the last metric update time of the startup event, delete event, update event, or any event.

You and I know what it stands for because we have context on it. That context should be captured in the code preferably with the variable name. Maybe something like updateEventSentTime